nixos/modules/services/hardware/amdgpu-fan.nix

```diff
@@ -0,0 +1,73 @@
+{ config, lib, pkgs, ... }:
+with lib;
+
+let
+  cfg = config.services.amdgpu-fan;
+  pkg = pkgs.amdgpu-fan;
+  amdgpuFanConfigFile = pkgs.writeText "amdgpu-fan.yml" ''
+    # speed matrix
+    # -[temp(*C), speed(0-100%)]
+    ${cfg.speed_matrix}
+
+    # optional list of cards
+    ${cfg.cards}
+  '';
+
+in
+{
+
+  options.services.amdgpu-fan = {
+    enable = mkEnableOption "Python daemon for controlling the fans on amdgpu cards";
+
+    speed_matrix = mkOption {
+      type = types.lines;
+      default = ''
+        - [0, 0]
+        - [40, 30]
+        - [60, 50]
+        - [80, 100]
+      '';
+      example = ''
+        - [0, 0]
+        - [20, 20]
+        - [40, 60]
+        - [60, 100]
+      '';
+      description = ''
+            Define fan speed for different temperatures using a speed matrix
+      '';
+    };
+
+    cards = mkOption {
+      type = types.lines;
+      default = "";
+      example = ''
+        cards:
+        - card0
+      '';
+      description = ''
+            Define cards that can be controlled. This can be any card returned from:
+            `ls /sys/class/drm | grep "^card[[:digit:]]$"`
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ pkg ];
+    environment.etc.amdgpuFanConfigFile.source = amdgpuFanConfigFile;
```

Comment:
is this the right location?
```
no config found, creating one in /etc/amdgpu-fan.yml
Traceback (most recent call last):
  File "/nix/store/c3vq1vdl7psic05vrb1dv0q39spkc6fg-python3.7-amdgpu-fan-0.0.6/bin/.amdgpu-fan-wrapped", line 9, in <module>
    sys.exit(main())
  File "/nix/store/c3vq1vdl7psic05vrb1dv0q39spkc6fg-python3.7-amdgpu-fan-0.0.6/lib/python3.7/site-packages/amdgpu_fan/controller.py", line 72, in main
    with open(CONFIG_LOCATIONS[-1], 'w') as f:
PermissionError: [Errno 13] Permission denied: '/etc/amdgpu-fan.yml'
```

Reply:
oh - I reused the variable (amdgpuFanConfigFile) instead of the file name